tests/common/fixtures/elasticsearch.py

```diff
@@ -0,0 +1,55 @@
+# -*- coding: utf-8 -*-
+from __future__ import unicode_literals
+
+import os
+
+import pytest
+
+from h.search import get_client
+from h.search import init
+
+ELASTICSEARCH_HOST = os.environ.get("ELASTICSEARCH_HOST", "http://localhost:9200")
+ELASTICSEARCH_INDEX = "hypothesis-test"
+
+
+@pytest.fixture
+def es_client():
+    """A :py:class:`h.search.client.Client` for the test search index."""
+    return _es_client()
```

Comment:
It's just because there's three separate places in this file where `_es_client()` is called. The `init_elasticsearch` fixture can't use the `es_client` fixture as a fixture. A session-scoped pytest fixture like `init_elasticsearch` (called only once per test session) can't depend on a test-scoped fixture like `es_client` (called once for each individual test that uses it). So instead of depending on the `es_client` fixture `init_elasticsearch` calls the `_es_client()` function instead.

I guess this also has the advantage that both `init_elasticsearch` and `delete_all_elasticsearch_documents` create their own client, rather than using the `es_client` fixture that the test might have messed with. (Though not likely any test would mess with it.)